src/setup-beam.js

```diff
@@ -620,13 +653,26 @@ function parseVersionFile(versionFilePath0) {
       }
     }
   })
-  if (!appVersions.size) {
-    core.info('There was apparently nothing to consume')
-  } else {
-    core.info('... done!')
-  }
-  core.endGroup()
+  return appVersions
+}
 
+function parseMiseVersionFile(fileData) {
+  const appVersions = new Map()
+  const toml = TOML.parse(fileData)
+  for (let app of ['erlang', 'elixir', 'gleam', 'rebar']) {
+    let version = toml.tools && toml.tools[app]
+    if (Array.isArray(version)) {
+      version = version[0]
+    }
+    if (typeof version === 'object' && version.version) {
```

Comment:
I'd add a comment here: "In case of a version object, we look for a specific key - version".

Reply:
Also, is this documented in the `mise` format? It'd be nice to add a comment (PR only, not code) here, for future reference.